rllib/examples/metrics/custom_metrics_in_env_runners.py

```diff
@@ -0,0 +1,339 @@
+"""Example of adding custom metrics to the results returned by `EnvRunner.sample()`.
+
+We use the `MetricsLogger` class, which RLlib provides inside all its components (only
+when using the new API stack through
+`config.api_stack(_enable_rl_module_and_learner=True,
+_enable_env_runner_and_connector_v2=True)`),
+and which offers a unified API to log individual values per iteration, per episode
+timestep, per episode (as a whole), per loss call, etc..
+`MetricsLogger` objects are available in all custom API code, for example inside your
+custom `Algorithm.training_step()` methods, custom loss functions, custom callbacks,
+and custom EnvRunners.
+
+This example:
```

Comment:
Awesome example!! This shows a lot how to use the `MetricsLogger`. 

To increase complexity, we could:
1. Run this only in evaluation and run evaluation each `n`th training step.
2. Reset some metrics in between.
3. Log two group of metrics (for two groups of env-runners).

Reply:
Good point! There are two more PRs in flight: MetricsLogger on algorithm.training_step and MetricsLogger inside loss function. A third one could be: MetricsLogger only on eval env runners 🙌